### ISSUE TITLE
Create a page to visualize Trello comments of a user

### ISSUE THREAD

**Issue by colingrodecoeur360** [360Learning/360learning.github.io#15]
# Context 

Flavien introduced his _MAgic PERformance Review Method_ (MAPERM) in this course: https://team.360learning.com/course/play/60e4664e08a16a9045addcbd

Key part:

> The MAPERM’s secret is to make a detailed assessment of each interaction, even the most insignificant one. There are two steps to achieve this:
-list all interactions;
-for each interaction, assess it.

Unfortunately, Trello's interface makes it really difficult to gather all the messages of a given person during some time period.
I figured it would be useful to use the Trello API to do that and display the results in a basic table.

# Spec

- Add a new `trelloCommentsDashboard` tool, drawing inspiration from [Adrien's Trello board building helper](https://360learning.github.io/trelloOnboardingBuilder.html)
- Form fields: 
  - Trello identifier
  - Auth token
  - The reviewee's Trello username
- Display a table with 4 columns: 
  - Date
  - Board name
  - Card name (with a link to the comment)
  - Comment
- To go further, add filters (dates, number of results, exclude some boards...) and sorting on each column

**Comment by colingrodecoeur360** [comment]
@AdriAt360 It was easier than I expected (and even quite fun) thanks to your previous work 🙏 

Can you perform a functional review? 

- Checkout `dev-trello-comment-dashboard` ([here](https://github.com/360Learning/360learning.github.io/pull/16) is the draft PR if you want to have a look)
- Open the page and follow your instincts (I think it's better if I don't give you detailed indications, let's see if the tool is intuitive enough)

**Comment by AdriAt360** [comment]
@colingrodecoeur360 that's nice 😍 
I like the UX, even though I think it can be improved (feel free to not do everything depending on the feasibility/effort)
- Start date/End date
  - I guess the main usage will be for perf reviews
  - would it be easy to automatically select the complete previous quarter instead of the last 90 days 
  - e.g. on April, 19, select "start date = 108" and "end date = 19" to match the range 2022-01-01 -> 2022-03-31
- sorting
  - is it easy to highlight the selected sorting?
  - e.g. below would be "sorted by descending date" (inspiring from https://unicode-table.com/en/sets/arrow-symbols/#up-arrows, but feel free to choose different icons)
![image](https://user-images.githubusercontent.com/54360965/163972391-a77b1f3d-86fa-4b3a-803a-819b3003252d.png)
- Truncate comments to N lines
  - lines can be huge
  - and I guess it would be more relevant to have similar sizing of the content
  - would it be easy to do "Truncate comments to X chars" instead (with a default to 250 maybe)
- Their username
  - I don't find the "their" very explicit, what would you think about "Colleague's username" maybe?
- I must admit I missed the subtitle at first, because it seems out of the page
![image](https://user-images.githubusercontent.com/54360965/163974779-b65512cc-7ef9-417b-9349-46d03b591793.png)
- To get started, get your Trello API information and store carefully your identifier and authentication token.
  - it would be even better if it's be explicit that this has to be done only once and then the credentials can be stored and reused
  - this is my fault on this part as it wasn't explicit in the other tool (by the way, feel free to change the wording there as well 😇 )
  - by the way I added such information inside [the linked doc](https://docs.google.com/document/d/1HwaedNa861gkj93TaradW5n0MID8cbeamiU5SXCvX64/edit#heading=h.ijjo2dol5z6v) (and granted you edit permission if you want to update this doc)

**Comment by colingrodecoeur360** [comment]
@AdriAt360 

Thanks for the feedback! I marked with a ➡️ the points that I changed below.

---

> would it be easy to automatically select the complete previous quarter instead of the last 90 days

I thought of it, but I had trouble finding a rule that worked well, I think what you propose is fine given that perf reviews will be the most common usage. ➡️ Can you check the change? 

---

> is it easy to highlight the selected sorting?

I could, but I prefer not spending time on that, the ROI seems low

---

> would it be easy to do "Truncate comments to X chars" instead (with a default to 250 maybe)

Actually, that's what I had done at first, but:
- I felt like the result was quite poor (comments truncated in the middle of a sentence or even a word)
- There is no guarantee that we would end up with similarly sized blocks, because of line breaks

Ok for you to leave it like that for now?

---

> I don't find the "their" very explicit, what would you think about "Colleague's username" maybe?

Ok I was not super fan either ➡️ I changed it

---

> I must admit I missed the subtitle at first, because it seems out of the page

Ok! Good to know that it's not visible enough like that ➡️ It's updated

---

> it would be even better if it's be explicit that this has to be done only once and then the credentials can be stored and reused

I agree, I did want to propose to improve the doc 👍 
We could explain how to store them in LastPass (Section "Save password manually" [here](https://support.lastpass.com/help/add-a-new-site-lp010005)). I will ping you on a rewrite draft once it's done

**Comment by AdriAt360** [comment]
@colingrodecoeur360 awesome =)
I agree with your choices, let's go prod 🕺🏽

**Comment by colingrodecoeur360** [comment]
Hey @FabriceGaudin 👋  I worked on this little side project, and I would like Flavien to review it, but he doesn't seem to be a member of that repo. Do you think you could give him commenting rights?

**Comment by FabriceGaudin** [comment]
@colingrodecoeur360 Great side project! 🎉 
Sure, I gave commenting rights to the whole GH Support team (https://github.com/orgs/360Learning/teams/support). Let me know if that doesn't work

**Comment by colingrodecoeur360** [comment]
@flavien360 As discussed I would love to have your feedback on this little tool 🙏 

Here is what you need to do:

- In a terminal, clone the repository : `git clone https://github.com/360Learning/360learning.github.io`
- `cd 360learning.github.io`
- `git checkout dev-trello-comment-dashboard`
- `open trelloCommentsDashboard.html`
- Enjoy ✨ 

(Let me know if you have suggestions on how to broadcast the tool as well - create a dedicated course, update yours...)

**Comment by colingrodecoeur360** [comment]
@AdriAt360 fyi I ended up refining the date management (absolute dates instead of "xx days ago"), if you want to give it a try

**Comment by flavien360** [comment]
@colingrodecoeur360 This tool is amazing! 😱

> I would love to have your feedback on this little tool

- I like the current format of dates, I think they are more usable than in https://github.com/360Learning/360learning.github.io/issues/15#issuecomment-1102396757. 👍
- I would suggest having a filtering field. Ideally, any kind of text; otherwise mentions (especially, mentioning me). I want to see the interactions the reviewee has had with me, primarily.
- A nice-to-have would be a system of tracking the items that have been handled (like a todo list). This may be too difficult to do, though.

> Let me know if you have suggestions on how to broadcast the tool as well - create a dedicated course, update yours...

I would highly suggest that we use it, before we communicate. I’ve observed adoption is tightly related to usage; a heuristic [used by Superhuman](https://firstround.com/review/how-superhuman-built-an-engine-to-find-product-market-fit/) is to answer the question "How would you feel if you could no longer use this tool?". I’d want to be "very disappointed" before sharing it. 🙂

**Comment by AdriAt360** [comment]
@colingrodecoeur360 great, I approve!
@flavien360 I like your suggestions too, even though it's not 100% clear what you have in mind for "items that have been handled". Not asking you to develop right now, but if you end up having both time and willing to write something down, I'd say opening a new ticket might be 100% welcome =)
And thanks for sharing your "very disappointed" tip and reference 🙏🏽

**Comment by colingrodecoeur360** [comment]
@flavien360 Thanks a lot!

> I would suggest having a filtering field.

I had thought about it for a next version, but you're right, it's actually a must-have. I added one, you can check the result by updating the branch (`git pull`):

<img width="792" alt="image" src="https://user-images.githubusercontent.com/51365591/164439776-cbd3da6d-cfd8-4b83-a189-4d3e0bacb9ed.png">

EDIT: it searches in the `Board`, `Card` and `Comment` columns, taking into account the full text (not just the truncated part of the comment), in a non-case-sensitive way.

> A nice-to-have would be a system of tracking the items that have been handled (like a todo list). This may be too difficult to do, though.

Indeed, it would be more difficult, as we would need some kind of database to store which items have been handled already.
But I keep your idea in mind, maybe I'll think of a simple trick to reach this goal.

> I would highly suggest that we use it...

Interesting take, I agree we shouldn't share it too fast if it doesn't provide a clear benefit 👍 
I'm confident it will, at least for me: I've already used it quite a lot in the last few days. But let's wait for the organic feedback of the first users of this tool!

**Comment by colingrodecoeur360** [comment]
@flavien360 @AdriAt360 It's alive 🚀 

https://360learning.github.io/trelloCommentsDashboard.html

**Comment by flavien360** [comment]
@colingrodecoeur360 

I tried to `git pull`, but got the following error, can you help? 😬

![CleanShot 2022-04-21 at 15 19 55](https://user-images.githubusercontent.com/46931903/164466856-f999b6e6-27ba-4476-8f4e-916438d1d9d4.png)

**Comment by colingrodecoeur360** [comment]
@flavien360 Ah yes that's because the branch has been deleted. If you want to fetch the latest changes on your machine, you'll have to use the `master` branch (`git checkout master && git pull`).

But anyway I don't recommend that, it will be easier for you to play with [the live version](https://360learning.github.io/trelloCommentsDashboard.html), now that it's deployed :)

**Comment by flavien360** [comment]
@colingrodecoeur360 All good, thanks!
